test/unit/util/test_find_dataset.py

```diff
@@ -0,0 +1,24 @@
+import os
+import tempfile
+
+from galaxy.util import find_dataset
+
+
+def test_find_dataset_no_dir():
+    assert find_dataset('no-such-path') is None
+
+
+def test_find_dataset_no_file():
+    td = tempfile.TemporaryDirectory()
+    td1 = tempfile.TemporaryDirectory(dir=td.name)
+    tempfile.mkstemp(dir=td1.name, suffix='.not-dat')
+    assert find_dataset('no-such-path') is None
```

Comment:
```suggestion
    with tempfile.TemporaryDirectory() as td_name:
        td1 = tempfile.TemporaryDirectory(dir=td_name)
        tempfile.mkstemp(suffix='.not-dat', dir=td1.name)
        assert find_dataset(td_name) is None
```

Reply:
@nsoranzo thanks! (my assert stmt made no sense calling the function with that argument!)